sermant-plugins/sermant-router/router-common/src/main/java/io/sermant/router/common/xds/lb/XdsLoadBalancerFactory.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2024-2024 Sermant Authors. All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package io.sermant.router.common.xds.lb;
+
+import io.sermant.core.service.ServiceManager;
+import io.sermant.core.service.xds.XdsCoreService;
+import io.sermant.core.service.xds.XdsLoadBalanceService;
+import io.sermant.core.service.xds.entity.XdsLbPolicy;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * XdsLoadBalancerFactory
+ *
+ * @author daizhenyu
+ * @since 2024-08-30
+ **/
+public class XdsLoadBalancerFactory {
+    private static final Map<String, XdsLoadBalancer> LOAD_BALANCERS = new ConcurrentHashMap<>();
+
+    private static final String RANDOM = "RANDOM";
+
+    private static XdsLoadBalanceService loadBalanceService = ServiceManager.getService(XdsCoreService.class)
+            .getLoadBalanceService();
```

Comment:
Be careful of the NPE thrown in the situation when XdsCoreService is not enabled

Reply:
done